include/dd/ComplexTable.hpp

```diff
@@ -135,70 +144,45 @@ namespace dd {
             if (std::abs(val - 1.) < TOLERANCE)
                 return &one;
 
+            if (std::abs(val - SQRT2_2) < TOLERANCE)
+                return &sqrt2_2;
+
             lookups++;
 
             // search in intended bucket
             const auto key = hash(val);
 
-            // ensure that the exact value of important constants is checked first on all occasions
-            if (key == zeroKey) {
-                if (std::abs(val) < TOLERANCE) {
-                    ++hits;
-                    return &zero;
-                } else {
-                    ++collisions;
-                }
-            } else if (key == oneKey) {
-                if (std::abs(val - 1.) < TOLERANCE) {
-                    ++hits;
-                    return &one;
-                } else {
-                    ++collisions;
-                }
-            } else if (key == sqrt2_2Key) {
-                if (std::abs(val - SQRT2_2) < TOLERANCE) {
-                    ++hits;
-                    return &sqrt2_2;
-                } else {
-                    ++collisions;
+            {
+                Entry* p = find(table[key], val);
+                if (p != nullptr) {
+                    return p;
                 }
             }
 
-            const auto& bucket = table[key];
-            auto        p      = find(bucket, val);
-            if (p != nullptr) {
-                return p;
-            }
-
             // search in (potentially) lower bucket
             if (val - TOLERANCE >= 0) {
                 const auto lowerKey = hash(val - TOLERANCE);
                 if (lowerKey != key) {
-                    const auto& lowerBucket = table[lowerKey];
-                    p                       = find(lowerBucket, val);
+                    ++neighbors;
+                    Entry* p = find(table[lowerKey], val);
```

Comment:
```suggestion
                    auto p = find(table[lowerKey], val);
```
consider using `auto` for staying consistent with the above statement

Reply:
Does not apply anymore.